### PR TITLE
added Tv247.us

### PR DESCRIPTION
**Pull request by hexated**

### CODE REVIEW

File: Tvtwofourseven/src/main/kotlin/com/lagradost/Tvtwofourseven.kt [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        val homePageList = ArrayList<HomePageList>()
+        listOf(
+            Pair("$mainUrl/top-channels", "Top Channels"),
+            Pair("$mainUrl/all-channels", "All Channels")
+        ).apmap {
+            val home =
+                app.get(it.first).document.select("div.grid-items div.item").mapNotNull { item ->
+                    item.toSearchResult()
+                }
+            if (home.isNotEmpty()) homePageList.add(HomePageList(it.second, home, true))
```
Comment (LagradOst) [inline Tvtwofourseven/src/main/kotlin/com/lagradost/Tvtwofourseven.kt:29]: this is simply wrong, it does not have a "next".

I would prefer if you instead used smth like

```kotlin
val list = listOf(
            Pair("$mainUrl/top-channels", "Top Channels"),
            Pair("$mainUrl/all-channels", "All Channels")
        ).apmap { (url,name) -> 
            val home =
                app.get(url).document.select("div.grid-items div.item").mapNotNull { item ->
                    item.toSearchResult()
                }
            HomePageList(name, home)
        }.filter { it.isNotEmpty() }
```

File: Tvtwofourseven/src/main/kotlin/com/lagradost/Tvtwofourseven.kt [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        }
+        return HomePageResponse(homePageList)
+    }
+
+    private fun Element.toSearchResult(): LiveSearchResponse? {
+        return LiveSearchResponse(
+            this.selectFirst("div.layer-content a")?.text() ?: return null,
+            fixUrl(this.selectFirst("a")!!.attr("href")),
```
Comment (LagradOst) [inline Tvtwofourseven/src/main/kotlin/com/lagradost/Tvtwofourseven.kt:37]: ?: return null

File: Tvtwofourseven/src/main/kotlin/com/lagradost/Tvtwofourseven.kt [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            "$mainUrl/wp-admin/admin-ajax.php", data = mapOf(
+                "action" to "ajaxsearchlite_search",
+                "aslp" to query,
+                "asid" to "1",
+                "options" to "qtranslate_lang=0&set_intitle=None&set_incontent=None&set_inposts=None"
+            ),
+            headers = mapOf("X-Requested-With" to "XMLHttpRequest")
+        ).document.select("div.item").mapNotNull {
+            LiveSearchResponse(
+                it.selectFirst("a")?.text() ?: return null,
```
Comment (LagradOst) [inline Tvtwofourseven/src/main/kotlin/com/lagradost/Tvtwofourseven.kt:56]: return@mapNotNull right?

**Review by LagradOst — CHANGES_REQUESTED** [review]

**Comment by hexated** [comment]
ready to merge

**Pull request merged**